crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -1,71 +1,349 @@
-	#[clap(long, conflicts_with = "execute")]
+	#[clap(short = 'D', long, conflicts_with = "execute")]
 	dry_run: bool,
+	/// Enables developer mode, bypassing certain user prompts for faster testing.
+	/// Recommended for testing and local development only.
+	#[clap(name = "dev", long, short, default_value = "false")]
+	dev_mode: bool,
 }
-
 impl CallContractCommand {
 	/// Executes the command.
-	pub(crate) async fn execute(self) -> anyhow::Result<()> {
-		clear_screen()?;
-		intro(format!("{}: Calling a contract", style(" Pop CLI ").black().on_magenta()))?;
-		set_theme(Theme);
+	pub(crate) async fn execute(mut self) -> Result<()> {
+		// Check if message specified via command line argument.
+		let prompt_to_repeat_call = self.message.is_none();
+		// Configure the call based on command line arguments/call UI.
+		if let Err(e) = self.configure(&mut cli::Cli, false).await {
+			match e.to_string().as_str() {
+				"Contract not deployed." => {
+					display_message(
+						"Use `pop up contract` to deploy your contract.",
+						true, // Not an error, just a message.
+						&mut cli::Cli,
+					)?;
+				},
+				_ => {
+					display_message(&e.to_string(), false, &mut cli::Cli)?;
+				},
+			}
+			return Ok(());
+		};
+		// Finally execute the call.
+		if let Err(e) = self.execute_call(&mut cli::Cli, prompt_to_repeat_call).await {
+			display_message(&e.to_string(), false, &mut cli::Cli)?;
+		}
+		Ok(())
+	}
+
+	fn display(&self) -> String {
+		let mut full_message = "pop call contract".to_string();
+		if let Some(path) = &self.path {
+			full_message.push_str(&format!(" --path {}", path.display()));
+		}
+		if let Some(contract) = &self.contract {
+			full_message.push_str(&format!(" --contract {}", contract));
+		}
+		if let Some(message) = &self.message {
+			full_message.push_str(&format!(" --message {}", message));
+		}
+		if !self.args.is_empty() {
+			let args: Vec<_> = self.args.iter().map(|a| format!("\"{a}\"")).collect();
+			full_message.push_str(&format!(" --args {}", args.join(", ")));
+		}
+		if self.value != DEFAULT_PAYABLE_VALUE {
+			full_message.push_str(&format!(" --value {}", self.value));
+		}
+		if let Some(gas_limit) = self.gas_limit {
+			full_message.push_str(&format!(" --gas {}", gas_limit));
+		}
+		if let Some(proof_size) = self.proof_size {
+			full_message.push_str(&format!(" --proof_size {}", proof_size));
+		}
+		full_message.push_str(&format!(" --url {} --suri {}", self.url, self.suri));
+		if self.execute {
+			full_message.push_str(" --execute");
+		}
+		if self.dry_run {
+			full_message.push_str(" --dry_run");
+		}
+		full_message
+	}
 
-		let call_exec = set_up_call(CallOpts {
+	/// Ensures the contract is ready by checking if it has been built. If not built, it builds the
+	/// contract. After building, prompts the user to confirm if the contract has already been
+	/// deployed.
+	async fn ensure_contract_ready(&self, cli: &mut impl Cli) -> Result<()> {
```

Comment:
I believe it would be nice doing a small refactor to separate the logic checking if the contract has been built from the one prompting the user if the contract has been deployed into separate functions.
`ensure_contract_ready` is not that clear to me, while if I see `ensure_contract_built` , I certainly know what to expect when reading through this code.

I appreciate that this was concretely included for this fix.

Reply:
Nice discussion around this refactor. Improved it in https://github.com/r0gue-io/pop-cli/pull/306/commits/b0f3e8b28b930a66353faf79166276ba48b9ec88

Reply:
Thanks for the quick action! I'll drop a review shortly :pray: